### PR TITLE
Surface Minio FS errors better in UI and support bundles

### PR DESCRIPTION
**Pull request by divolgin**
#### What type of PR is this?

type::bug
#### What this PR does / why we need it:

When NFS backend configuration for snapshots fails, the user is presented with a generic error message. The specific error is also no surfaced by support bundle analyzers.

This will also clean up orphaned `kotsadm-fs-minio-check` pods once configuration is successful.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

- This probably doesn't cover every possible failure mode.
- UI could use some work for cases where error messages are excessively long.
- Screenshots in comments.

## Steps to reproduce

1. Navigate to Snapshots settings page
2. Trigger error in one or more of the following ways:
   - enter invalid IP address/host name (invalid will cause error to surface sooner; i/o timeout takes longer and may not happen fast enough)
   - remove/rename `mount.nfs` binary on the host
   - remove write permissions from the NFS directory
3. Click "Update" and wait
4. Collect support bundle.

#### Does this PR introduce a user-facing change?

```release-note
* Failed `kotsadm-fs-minio-check` pods will now be cleaned up once NFS backend for snapshots has been configured successfully.
* Adds collector and analyzer for cases when NFS configuration fails because the `mount.nfs` binary is missing on the host.
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

**Comment by divolgin** [comment]
Some screenshots:
Screen Shot 2022-05-06 at 12 40 41 PM
Screen Shot 2022-05-06 at 6 02 51 PM
Screen Shot 2022-05-06 at 5 59 33 PM
Screen Shot 2022-05-06 at 5 42 03 PM
Screen Shot 2022-05-06 at 3 39 06 PM
Screen Shot 2022-05-06 at 2 39 51 PM

File: pkg/snapshot/filesystem.go [hunk 1]
```diff
@@ -41,6 +45,12 @@ const (
 	FileSystemMinioServicePort                                                = 9000
 )
 
+const (
+	fdMinioCheckTage  = "kotsadm-fs-minio-check"
```
Comment (sgalsaleh) [inline pkg/snapshot/filesystem.go:49]:
```suggestion
	fsMinioCheckTag  = "kotsadm-fs-minio-check"
```

File: pkg/snapshot/filesystem.go [hunk 2]
```diff
@@ -548,7 +558,40 @@ func shouldResetFileSystemMount(ctx context.Context, clientset kubernetes.Interf
 	}
 
 	if err := k8sutil.WaitForPod(ctx, clientset, deployOptions.Namespace, checkPod.Name, time.Minute*2); err != nil {
-		finalErr = errors.Wrap(err, "failed to wait for file system minio check pod to complete")
+		genericErr := errors.Wrap(err, "failed to wait for file system minio check pod to complete")
```
Comment (sgalsaleh) [inline pkg/snapshot/filesystem.go:561]: what about the reset pod and the keys SHA pod? i see this PR will only return errors for the check pod

Comment (divolgin) [reply pkg/snapshot/filesystem.go:561]: I think those will still show a generic error and logs will be in the support bundle....  This is the only one I tested.

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**